app/models/staff.rb

```diff
@@ -27,35 +27,36 @@ def powers
     canonical_staff&.powers || Power.none
   end
 
+  def canonical_model
+    canonical_staff
+  end
+
   def canonical_models
-    return Canonical::Staff.where(id: canonical_staff.id) if canonical_staff.present?
+    return Canonical::Staff.where(id: canonical_staff_id) if canonical_model_matches?
 
-    canonicals = Canonical::Staff.where('name ILIKE ?', name)
+    query = 'name ILIKE :name'
+    query += ' AND (magical_effects ILIKE :magical_effects)' unless magical_effects.nil?
```

Comment:
Since `magical_effects` can be lengthy, it'd be good to not require users to capitalise perfectly.